scripts/DP28-JP/c100428036.lua

```diff
@@ -0,0 +1,68 @@
+--BK キング・デンプシー
+
+--Script by Chrono-Genex
+function c100428036.initial_effect(c)
+	--xyz summon
+	aux.AddXyzProcedure(c,nil,4,2)
+	c:EnableReviveLimit()
+	--to hand
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(100428036,0))
+	e1:SetCategory(CATEGORY_TOHAND+CATEGORY_SEARCH+CATEGORY_TOGRAVE)
+	e1:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e1:SetProperty(EFFECT_FLAG_DELAY)
+	e1:SetCode(EVENT_SPSUMMON_SUCCESS)
+	e1:SetCountLimit(1,100428036)
+	e1:SetTarget(c100428036.thtg)
+	e1:SetOperation(c100428036.thop)
+	c:RegisterEffect(e1)
+	--cannot be target
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(100428036,1))
+	e2:SetType(EFFECT_TYPE_QUICK_O)
+	e2:SetCode(EVENT_FREE_CHAIN)
+	e2:SetRange(LOCATION_MZONE)
+	e2:SetCountLimit(1,100428036+100)
+	e2:SetHintTiming(0,TIMINGS_CHECK_MONSTER)
+	e2:SetTarget(c100428036.tgtg)
+	e2:SetOperation(c100428036.tgop)
+	c:RegisterEffect(e2)
+end
+function c100428036.thfilter(c)
+	return (c:IsLevelBelow(4) and c:IsRace(RACE_WARRIOR) and c:IsAttribute(ATTRIBUTE_FIRE)
+		or c:IsSetCard(0x2084) and c:IsType(TYPE_SPELL+TYPE_TRAP)) and (c:IsAbleToHand() or c:IsAbleToGrave())
+end
+function c100428036.thtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100428036.thfilter,tp,LOCATION_DECK,0,1,nil) end
+end
+function c100428036.thop(e,tp,eg,ep,ev,re,r,rp)
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_OPERATECARD)
+	local g=Duel.SelectMatchingCard(tp,c100428036.thfilter,tp,LOCATION_DECK,0,1,1,nil)
+	local tc=g:GetFirst()
+	if tc and tc:IsAbleToHand() and (not tc:IsAbleToGrave() or Duel.SelectOption(tp,1190,1191)==0) then
+		Duel.SendtoHand(tc,nil,REASON_EFFECT)
+		Duel.ConfirmCards(1-tp,tc)
+	else
+		Duel.SendtoGrave(tc,REASON_EFFECT)
```

Comment:
redtext on line 46: `Parameter 1 should be "Card" or "Group".`
```suggestion
	elseif tc then
		Duel.SendtoGrave(tc,REASON_EFFECT)
```